layers/+lang/ess/packages.el

```diff
@@ -36,7 +36,6 @@
            ("\\.do\\'"           . STA-mode)
            ("\\.ado\\'"          . STA-mode)
            ("\\.[Ss][Aa][Ss]\\'" . SAS-mode)
-           ("\\.jl\\'"           . ess-julia-mode)
```

Comment:
now you are simply disabling julia for ess, so no one will be able to use ess with julia. Not sure what would be the best solution, but probably a flag like
`(defvar ess-use-julia t)`
and then disable stuff conditionally
then I can simply specify in my `.spacemacs`
```elisp
(setq-default
 dotspacemacs-configuration-layers
 (ess :variables
       ess-use-julia nil))
```

Reply:
When I get back to this, the plan is to move all of the julia-related ESS stuff into the julia layer. This change was why I marked this PR as WIP since using my branch could very easily break workflows until I finish that migration. So `.jl` files will still open in `ess-julia-mode` (for those who customize this layer to do so), but you'll have to install the julia layer for that rather than the ESS layer.

Reply:
I am not sure, where you are going with this. Please don't break my ESS julia setup. ESS has quite decent Julia support out of the box for now (which does not mean it cannot be improved) and this should be unchanged if only the `ess` layer is installed.

Reply:
If you want it to be, your experience will be identical; you'll just have to add the julia layer in addition to the ESS layer. If ESS provides functionality that's only applicable to julia (it does), then it makes no sense to keep it in the generic ESS layer instead of moving it to a layer specifically for julia.

Reply:
The thing with `ESS` (as in the emacs package to use R/STATA/Julia/S/..., in contrast to `ess`, the layer for spacemacs) is its age, on the one hand it has a lot of features and on the other it has a lot of ancient code no one has touched for ages. The current maintainers are all R users (this is my personal impression), so there is still quite a bit of ongoing development for R. Julia support was somewhat put on top of that, integration is far from perfect and would probably require a bit of work and dealing with the `ESS` codebase. So I am not sure if it would be better to start fresh, or to stick to `ESS`.

By your logic there should be an `r` layer instead of an `ess` layer, so I think that the `ess` layer should keep its ability to be used with julia out of the box.

Questions: 
1) Which parts of `ess` are julia only? 
2) you are going to use `julia-repl`, how is this going to give an identical experience?